Makefile

```diff
@@ -0,0 +1,16 @@
+all: doc example demo
```

Comment:
I'd say any Makefile which does more than one thing (i.e. has more than one target) should have help as default operation. The help shouldn't be verbose - if target names are self-explanatory, then the whole help could be a one liner with invocation syntax listing all targets.